PowerEditor/src/ScintillaComponent/Buffer.cpp

```diff
@@ -721,7 +721,18 @@ bool FileManager::reloadBuffer(BufferID id)
 	buf->setLoadedDirty(false);	// Since the buffer will be reloaded from the disk, and it will be clean (not dirty), we can set _isLoadedDirty false safetly.
 								// Set _isLoadedDirty false before calling "_pscratchTilla->execute(SCI_CLEARALL);" in loadFileData() to avoid setDirty in SCN_SAVEPOINTREACHED / SCN_SAVEPOINTLEFT
 
+	bool bUnsyncStateBeforeReload = buf->isUnsync(); // store for a possible unsync state restoration later 
+	buf->setUnsync(false);	// we have to ensure that isUnsync is false before we try to call the loadFileData below for the reloading
+							// - after a successful reload, the source file and its filebuffer in N++ are the same, so the isDirty should be false (and the isUnsync too)
+							// - but when the isUnsync is true at this point, any successfully reloaded filebuffer later will end up marked as dirty!
+							//   (similar reason as in the above setLoadedDirty(false), but now it is the "_pscratchTilla->execute(SCI_SETSAVEPOINT)" in loadFileData(),
+							//    which ends up in the Notepad_plus::notify(), case SCN_SAVEPOINTLEFT, where is logic "if (buf->isUnsync()) isDirty = true")
+
 	bool res = loadFileData(doc, buf->getFullPathName(), data, &UnicodeConvertor, loadedFileFormat);
+
+	if (!res)
+		buf->setUnsync(bUnsyncStateBeforeReload);

```

Comment:
So it's enough to call setUnsync(true) once only when the `res` is true, isn't it?

Reply:
No.
When the `res` is true, reloading succeeded and the filebuffer `_isUnsync` and `_isDirty` should be both false.
So only when the `res` is false, I am taking care and restoring it to the previous state whatever it is.

Reply:
Sorry correction:
So it's enough to call setUnsync(f**alse**) once only when the res is true, isn't it?

Reply:
If res is true, then set unsync false unconditionally, otherwise, just do nothing.
It's not necessary to call it twice.

Reply:
There were in fact two problems, when the `_isUnsync` was true at this point:
- this flag caused incorrect activation of the _isDirty flag later in the loadFileData()
- nobody cared to reset it back to false after a successful reload

I solved both problems by calling `buf->setUnsync(false)` unconditionally before the loadFileData().
Only if the reloading in loadFileData() fails, one should care about resetting it back to a possible previous unsynced state.
So maybe I should rather use:

```
if (!res && bUnsyncStateBeforeReload)
	buf->setUnsync(true);
```
Is that form better?

Reply:
bUnsyncStateBeforeReload is useless IMO.
The fix can be much simpler:
```cpp
if (res)
    buf->setUnsync(false);
```
We can ignore current unsync state until reload result is true, then unsync state must be set to false. Or am I missing something?

Reply:
BTW, you can set inside of the following block:
```cpp
if (res)
	{
		setLoadedBufferEncodingAndEol(buf, UnicodeConvertor, loadedFileFormat._encoding, loadedFileFormat._eolFormat);
	}
	```

Reply:
> We can ignore current unsync state until reload result is true, then unsync state must be set to false. Or am I missing something?

Yes, it will not fix the issue, _isDirty will be triggered in the loadFileData().

It can be:

```
bool res = loadFileData(doc, buf->getFullPathName(), data, &UnicodeConvertor, loadedFileFormat);
if (res)
{
	buf->setUnsync(false);
	buf->setDirty(false);
}
```
But I will not put it in the 
```
if (res)
{
	setLoadedBufferEncodingAndEol(buf, UnicodeConvertor, loadedFileFormat._encoding, loadedFileFormat._eolFormat);
}
```
because of there is the `buf->_canNotify = true;` before it and I do not want to verify possible consequences.

So do you agree with this:

```
bool res = loadFileData(doc, buf->getFullPathName(), data, &UnicodeConvertor, loadedFileFormat);
if (res)
{
	buf->setUnsync(false);
	buf->setDirty(false);
}

buf->_canNotify = true;

if (res)
{
	setLoadedBufferEncodingAndEol(buf, UnicodeConvertor, loadedFileFormat._encoding, loadedFileFormat._eolFormat);
}

return res;
```

Reply:
@xomx 
> it will not fix the issue, _isDirty will be triggered in the loadFileData().

You are right about it 

Let's use the following version - it'll be more clear IMO. 
```cpp
bool res = loadFileData(doc, buf->getFullPathName(), data, &UnicodeConvertor, loadedFileFormat);
if (res)
{
	buf->setUnsync(false);
	buf->setDirty(false);
}

buf->_canNotify = true;

if (res)
{
	setLoadedBufferEncodingAndEol(buf, UnicodeConvertor, loadedFileFormat._encoding, loadedFileFormat._eolFormat);
}
```

Reply:
Ok.

Reply:
> because of there is the buf->_canNotify = true; before it and I do not want to verify possible consequences.

Just check it, you can safe to put 2 line into the block :
```
if (res)
{
	buf->setUnsync(false);
	buf->setDirty(false);
	setLoadedBufferEncodingAndEol(buf, UnicodeConvertor, loadedFileFormat._encoding, loadedFileFormat._eolFormat);
}
buf->_canNotify = true;
```
`buf->_canNotify = false;` is set in the beginning of function `reloadBuffer` for locking the loading period (especial the call of `loadFileData`) without notification. 

But if you don't want to do it, it's OK, I'll modify it afterward.

Reply:
I am not as familiar with the N++ code as you, so I was just careful not to screw up something. Modify the code as you think it fits. Good night.

Reply:
Thank you @xomx for your fix and consistence!